src/fractional-permissions.md

```diff
@@ -0,0 +1,159 @@
+# Permission to read
+
+So far we have seen permissions of the form `acc(x)` for a pointer `x`.
+We can be more specific and give a permission amount as the second argument to `acc`, as a fractional number.
+In this case we speak of _fractional permissions_:
+- A permission amount of `1` allows us to read and write (e.g. `acc(x, 1)`, or equivalently `acc(x)`, `acc(x, writePerm)`, `acc(x, 1/1)`)
+- A positive amount `< 1` allows only reading (e.g. `acc(x, 1/2)`, `acc(x, 3/4)`)
+- No access is denoted as `acc(x, 0)` or equivalently `acc(x, noPerm)`.
+- A permission amount `> 1` for a pointer is a contradiction and implies `false`. <!-- not for predicates -->
+
+Permission amounts to the same location can be summed up, for example `acc(x, 3/4)` is equivalent to `acc(x, 1/4) && acc(x, 1/4) && acc(x, 1/4)`.
+For concurrency this will be useful since it allows us to split read permissions to multiple threads and guaranteeing that there can only be one thread alive with exclusive write permission to a location [^1].
```

Comment:
this talks about concurrency, but not about framing. I would mention that, if you need to establish that the value stored in a location did not change, you would need to add a postcondition, which is cumbersome